src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritQueryHandlerHttp.java

```diff
@@ -0,0 +1,497 @@
+   *                           Meaning if --files should be appended to the command call.
+   * @param getCommitMessage   if full commit message should be included in the result.
+   *                           Meaning if --commit-message should be appended to the command call.
+   * @return the query result as a List of JSONObjects.
+   * @throws GerritQueryException if Gerrit reports an error with the query.
+   * @throws SshException         if there is an error in the SSH Connection.
+   * @throws IOException          for some other IO problem.
+   */
+  public List<JSONObject> queryJava(String queryString, boolean getPatchSets, boolean getCurrentPatchSet,
+                                    boolean getFiles, boolean getCommitMessage) throws SshException,
+      IOException, GerritQueryException {
+    return queryJava(queryString, getPatchSets, getCurrentPatchSet, getFiles, getCommitMessage, false);
+  }
+
+  //CS IGNORE RedundantThrows FOR NEXT 24 LINES. REASON: Informative.
+
+  /**
+   * Runs the query and returns the result as a list of Java JSONObjects.
+   *
+   * @param queryString        the query.
+   * @param getPatchSets       getPatchSets if all patch-sets of the projects found should be included in the result.
+   *                           Meaning if --patch-sets should be appended to the command call.
+   * @param getCurrentPatchSet if the current patch-set for the projects found should be included in the result.
+   *                           Meaning if --current-patch-set should be appended to the command call.
+   * @param getFiles           if the files of the patch sets should be included in the result.
+   *                           Meaning if --files should be appended to the command call.
+   * @param getCommitMessage   if full commit message should be included in the result.
+   *                           Meaning if --commit-message should be appended to the command call.
+   * @param getComments        if patchset comments should be included in the results.
+   *                           Meaning if --comments should be appended to the command call.
+   * @return the query result as a List of JSONObjects.
+   * @throws GerritQueryException if Gerrit reports an error with the query.
+   * @throws IOException          for some other IO problem.
+   */
+  public List<JSONObject> queryJava(String queryString, boolean getPatchSets, boolean getCurrentPatchSet,
+                                           boolean getFiles, boolean getCommitMessage, boolean getComments)
+      throws IOException, GerritQueryException {
+
+    List<JSONObject> list = new ArrayList<>();
+
+    runQuery(queryString, getPatchSets, getCurrentPatchSet, getFiles, getCommitMessage, getComments,
+        new GerritQueryHandlerHttp.LineVisitor() {
+          @Override
+          public void visit(String line) {
+            JSONObject json = (JSONObject)JSONSerializer.toJSON(line.trim());
+            list.add(json);
+          }
+        });
```

Comment:
die Methode runQuery ist ja private, da könnte man sich das LineVisitor Interface sparen und anstatt dessen einfach einen Consumer<String> übergeben und in der runQuery dann einfach in der Schleife visitor.accept(String) machen